CMakeLists.txt

```diff
@@ -46,11 +46,6 @@
 # Squelch a warning when building on Win32/Cygwin
 set(CMAKE_LEGACY_CYGWIN_WIN32 0)
 
-# Add a path for any locally-supplied CMake modules
-# These would typically be a part of any custom PSPs in use.
-# (this is not required, and the directory can be empty/nonexistent)
-set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/../psp/cmake/Modules" ${CMAKE_MODULE_PATH})
```

Comment:
I believe the setting of the CMAKE_MODULE_PATH must be done _before_ the call to the project() function or else it does not work.  So this does need to stay up here, I think.

Reply:
The reason why I have moved this line was because it was using the new `CFE_SOURCE_DIR` which was defined in the `initialize_globals` function which is called a few lines below.

I have pushed a change which uses `CMAKE_CURRENT_SOURCE_DIR` to overcome the fact that the `CFE_SOURCE_DIR` is undefined before the `project(CFE C)` gets called.

P.S. I am curious how do you verify that `set(CMAKE_MODULE_PATH...` does not work when set after the `project(...` call?

Reply:
This is relevant to the RTEMS build.  Cmake does not include an RTEMS platform module out of the box so one is included in the PSP.  In order for CMake to find it, it must be in the CMAKE_MODULE_PATH, which is searched at the time the "project" command is executed.

Reply:
Thanks for the answer. I have asked about the order because I don't see any connection between setting `CMAKE_MODULE_PATH` and `project` on their pages: 

> Semicolon-separated list of directories specifying a search path for CMake modules to be loaded by the include() or find_package() commands before checking the default modules that come with CMake. By default it is empty, it is intended to be set by the project. https://cmake.org/cmake/help/latest/variable/CMAKE_MODULE_PATH.html

and 

> Call the project() command near the top of the top-level CMakeLists.txt, but after calling cmake_minimum_required(). It is important to establish version and policy settings before invoking other commands whose behavior they may affect. See also policy CMP0000. https://cmake.org/cmake/help/latest/command/project.html?highlight=project

> In order for CMake to find it, it must be in the CMAKE_MODULE_PATH, which is searched at the time the "project" command is executed.

If this is some sort of implicit behavior and the `project` and `CMAKE_MODULE_PATH` in the same file should always go in the order like you describe I would suggest adding a comment because it is not obvious from the documentation. And I would not be able to test the RTEMS target at the moment.

Anyway after my latest change, I think that your original comment is satisfied.